bigtable/admin.go

```diff
@@ -1309,22 +1314,22 @@ func (iac *InstanceAdminClient) GetCluster(ctx context.Context, instanceID, clus
 		KMSKeyName:  kmsKeyName,
 	}
 	// Use type assertion to handle protobuf oneof type
-	if cfg, ok := c.Config.(*btapb.Cluster_ClusterConfig_); ok {
-		if asc, ok := autoscalingConfig(cfg); ok {
-			ci.AutoscalingConfig = asc
+	if cfg := c.GetClusterConfig(); cfg != nil {
+		if asc, ok := fromClusterConfigProto(cfg); ok {
+			ci.AutoscalingConfig = &asc
 		}
 	}
 	return ci, nil
 }
 
-func autoscalingConfig(c *btapb.Cluster_ClusterConfig_) (AutoscalingConfig, bool) {
-	if c.ClusterConfig == nil {
+func fromClusterConfigProto(c *btapb.Cluster_ClusterConfig) (AutoscalingConfig, bool) {
```

Comment:
Based on how you're using this, maybe simpler just to have this return *AutoscalingConfig which may be nil if it's not present in the proto?

Reply:
Yes, that's better and consistent with other similar functions. Done.